### PR TITLE
DetectorDescrition and Geometry Clang readability-container-size-empty

### PR DESCRIPTION
**Pull request by ianna**
* DetectorDescrition and Geometry part of @davidlange6's https://github.com/cms-sw/cmssw/pull/19661

### CODE REVIEW

**Comment by ianna** [comment]
please test

**Comment by ianna** [comment]
hold

**Comment by ianna** [comment]
unhold

**Comment by ianna** [comment]
please test

**Comment by ianna** [comment]
test with #20033

**Comment by ianna** [comment]
test with #20033

**Comment by ianna** [comment]
+1

**Comment by ianna** [comment]
As requested by @slava77 and @perrotta - the following PR is merged here https://github.com/cms-sw/cmssw/pull/20030

**Comment by ianna** [comment]
+1

File: DataFormats/Math/interface/GraphWalker.h [hunk 1]
```diff
@@ -70,8 +70,8 @@ GraphWalker<N,E>::GraphWalker(const Graph<N,E> & g)
  : graph_(g)
 {  // complexity = (no nodes) * (no edges)
    graph_.findRoots(root_);
-   stack_.push_back(edge_range(root_.begin(),root_.end())); 
-   if (root_.size()) {
+   stack_.emplace_back(edge_range(root_.begin(),root_.end())); 
+   if (root_.empty()) {
```
Comment (perrotta) [inline DataFormats/Math/interface/GraphWalker.h]: During last commit https://github.com/cms-sw/cmssw/pull/20020/commits/1e62fd6fe2504fa47becc9627d45d10308e498e0 the "!" was reverted here: please fix,

I didn't spot other examples of that kind in the DetectorDescription part of this PR, but please double check

**Comment by ianna** [comment]
Thanks for spotting this, I will check

**Comment by ianna** [comment]
please test

**Comment by ianna** [comment]
+1

**Comment by perrotta** [comment]
+1
- Technical
- No differences expected in outputs, no differences observed in jenkins tests

**Comment by davidlange6** [comment]
merge

**Pull request merged**